code/modules/clothing/head/_head.dm

```diff
@@ -27,4 +29,25 @@
 	..()
 	if(ismob(loc))
 		var/mob/M = loc
-		M.update_inv_head()
\ No newline at end of file
+		M.update_inv_head()
+
+/obj/item/clothing/head/throw_impact(atom/hit_atom, datum/thrownthing/thrownthing)
+	. = ..()
+	if(!.)
+		return
+	if(thrownthing && thrownthing.target_zone && thrownthing.target_zone != BODY_ZONE_HEAD)

```

Comment:
This should probably check if the hit atom is the throwing target or this might have weird side-effects.